### PR TITLE
Fixed get_weil_polys.

### PR DESCRIPTION
**Pull request by koffie**
This solves #16 

Note that before this commit we have
```
F = GF(7^3)
%time weil_polys = get_weil_polys(F)
```
gives:
```
CPU times: user 1min 56s, sys: 719 ms, total: 1min 57s
Wall time: 1min 57s
```
while after this commit we have:
```
CPU times: user 1.13 ms, sys: 0 ns, total: 1.13 ms
Wall time: 1.14 ms
```
so we also have the much wanted speedup.

With this enhancement we might want to make "loop_curves" the default.

### CODE REVIEW

**Comment by koffie** [comment]
Hi @BarinderBanwait , when you merge this commit could you do a squash and merge so that the last two silly commits won't stay in the history of the project forever.

File: isogeny_primes.py [hunk 1]
```diff
@@ -51,19 +51,42 @@
...
+    return False
+    
+def get_weil_polys(F):
+    """
+    Returns als degree 2 weil polynomials over F that are actually comming from an elliptic curve.
+    This uses theorem 4.1 of http://archive.numdam.org/article/ASENS_1969_4_2_4_521_0.pdf
+    """
+    q = F.characteristic()
+    a = F.degree()
+    R = PolynomialRing(QQ,'x')
```
Comment (BarinderBanwait) [inline isogeny_primes.py:86]: This is already defined as a global, since loads of methods need it, so it should be removed here.

**Review by BarinderBanwait — CHANGES_REQUESTED** [review]
Looks good ... I would suggest that the `q` be replaced with `p` throughout since then it corresponds exactly to Waterhouse Theorem 4.1.

**Review by BarinderBanwait — APPROVED** [review]
Looks good, will merge!

**Pull request merged**